cloudinit/config/cc_refresh_rmc_and_interface.py

```diff
@@ -0,0 +1,93 @@
+# (c) Copyright IBM Corp. 2020 All Rights Reserved
+#
+# Author: Aman Kumar Sinha <amansi26@in.ibm.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+"""
+Refresh IPv6 interface and RMC
+------------------------------
+**Summary:** Ensure Network Manager is not managing IPv6 interface
+
+This module is IBM PowerVM Hypervisor specific
+
+Reliable Scalable Cluster Technology (RSCT) is a set of software components
+that together provide a comprehensive clustering environment(RAS features)
+for IBM PowerVM based virtual machines. RSCT includes the Resource
+Monitoring and Control (RMC) subsystem. RMC is a generalized framework used
+for managing, monitoring, and manipulating resources. RMC runs as a daemon
+process on individual machines and needs creation of unique node id and
+restarts during VM boot.
+More details refer
+https://www.ibm.com/support/knowledgecenter/en/SGVKBA_3.2/admin/bl503_ovrv.htm
+
+This module handles
+- Refreshing RMC
+- Disabling Network Manager from handling IPv6 interface, as IPv6 interface
+  is used for communication between RMC daemon and PowerVM hypervisor.
+
+**Internal name:** per  always
+
+**Supported distros:** RHEL
+
+"""
+
+from cloudinit.settings import PER_ALWAYS
+from cloudinit import util
+from cloudinit import netinfo
+from cloudinit import subp
+
+frequency = PER_ALWAYS
+
+RMCCTRL = '/opt/rsct/bin/rmcctrl'
+SED = '/usr/bin/sed'
+
+
+def handle(name, _cfg, _cloud, log, _args):
+    info = netinfo.netdev_info()
+    # Setting NM_CONTROLLED=no for IPv6 interface
+    # making it down and up
+    info_len = len(info.keys())
+
+    log.debug('Making the ipv6 up')
+    for info_count in range(info_len):
+        try:
+            if 'lo' in list(info.keys())[info_count]:
+                log.debug('Skipping localhost interface')
+                continue
+            interface = list(info.keys())[info_count]
+            if not info[str(list(info.keys())[info_count])]['ipv4']:
+                refresh_ipv6(interface)
+                disable_NM_ipv6(interface)
+        except BaseException:
+            util.logexc(log, 'Interface is not up')
+
+    refresh_rmc(log)
+
+
+def refresh_ipv6(interface):
+    subp.subp(['ip', 'link', 'set', interface, 'down'])
+    subp.subp(['ip', 'link', 'set', interface, 'up'])
+
+
+def disable_NM_ipv6(interface):
```

Comment:
It will be good to add some comments on what is the relevance of each code block here. For eg. the whole intent of this module is to ensure that the IPV6 interface is always up and active, and the reason for that is the communication between the PowerVM hyp and RMC communication happens over this interface. It is also good to add some comments to indicate what happens if this communication is lost. For eg., if this communication is lost for some reason (which could happen if RHEL Network Manager manages this interface), then the RMC state of this VM, from the management plane perspective in inactive. When the RMC state of a VM is inactive, there are a lot of operations like VM migration (from one host to another) that is not supported by the Hypervisor.

Reply:
So the intend of this module is to ensure that the communication b/w the hyp and VM is not interrupted due to Network Manager. For this purpose, as part of this module, the NM is explicitly set to No for IPV6 interface and the interface is explicitly brought up, subsequent to which the RMC services are restarted to re-establish the communication with the hypervisor. 

It will be good if this information is explicitly covered as part of different comments.